### PR TITLE
Akshay

### PR DESCRIPTION
**Pull request by guytonde**
I learned about semaphores and mutexes.

### CODE REVIEW

File: Controls/Mail/Mailman.c [hunk 1]
```diff
@@ -13,6 +15,23 @@
...
+
+	OSSemPost(&MailboxFlag_Sem4, 
+			  OS_OPT_POST_1, 
+			  &err);
+
+	// Populate the mailbox
+	for (int i = 0; i < 256; i++) {
+		mailbox[i] = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"[random() % 26];
+	}
+	mailbox[256] = '\0';
```
Comment (Cam0Cow) [inline Controls/Mail/Mailman.c:27]: The mailbox variable is declared as `char mailbox[256]`. Would this assignment be an issue?

Comment (guytonde) [reply Controls/Mail/Mailman.c:27]: Which assignment? The one in the for loop or the null terminator assignment outside the loop?

Comment (Cam0Cow) [reply Controls/Mail/Mailman.c:27]: The null terminator

**Review by diyarajon — COMMENTED** [review]
Looks good, good job. Make sure to check out Roie's comment about the null terminator.

**Pull request closed without merging**